### PR TITLE
add validation for  StringMatch

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fixes: https://github.com/istio/istio/issues/48529 https://github.com/istio/istio/issues/48534

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
It lgtm. But i think you should validate not only method prefix, but also other fields too.

**Comment by zirain** [comment]
> It lgtm. But i think you should validate not only method prefix, but also other fields too.

you're right, do same to `scheme` and `authority`.

File: pkg/config/validation/virtualservice.go [hunk 1]
```diff
@@ -153,10 +153,17 @@ func validateHTTPRouteMatchRequest(http *networking.HTTPRoute, routeType HTTPRou
...
 				errs = appendErrors(errs, validateStringMatchRegexp(match.GetScheme(), "scheme"))
+				errs = appendErrors(errs, validateStringMatchPrefix(match.GetScheme(), "scheme"))
+
 				errs = appendErrors(errs, validateStringMatchRegexp(match.GetMethod(), "method"))
+				errs = appendErrors(errs, validateStringMatchPrefix(match.GetMethod(), "method"))
+
 				errs = appendErrors(errs, validateStringMatchRegexp(match.GetAuthority(), "authority"))
+				errs = appendErrors(errs, validateStringMatchPrefix(match.GetAuthority(), "authority"))
 				for _, qp := range match.GetQueryParams() {
 					errs = appendErrors(errs, validateStringMatchRegexp(qp, "queryParams"))
```
Comment (hzxuzhonghu) [inline pkg/config/validation/virtualservice.go:168]: also queryParam prefix check

**Review by hzxuzhonghu — COMMENTED** [review]
You missed updating `validateAllowOrigins`

**Comment by hzxuzhonghu** [comment]
I would suggest doing a refactor aswell, calling `validateStringMatchRegexp` from `validateStringMatch`, then in `validateStringMatch`, we validate regex/ exact and prefix

**Review by hzxuzhonghu — APPROVED** [review]
Thanks

**Pull request merged**